File: LagbajaSchools.java
import java.text.DecimalFormat;
import java.util.Scanner;

public class LagbajaSchools {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("How many students do you have? ");
        int numStudents = scanner.nextInt();
        System.out.print("How many subjects do they offer? ");
        int numSubjects = scanner.nextInt();
        System.out.println("Saving >>>>>>>>>>>>>>>>>>>>> saved successfully");

        int[][] scores = new int[numStudents][numSubjects];
        for (int count = 0; count < numStudents; count++) {
            System.out.println("Entering score for student " + (count + 1));
            for (int index = 0; index < numSubjects; index++) {
                while (true) {
                    System.out.print("Enter score for subject " + (index + 1) + ": ");
                    int score = scanner.nextInt();
                    if (score >= 0 && score <= 100) {
                        scores[count][index] = score;
                        System.out.println("Saving >>>>>>>>>>>>>>>>>>>>> saved successfully");
                        break;
                    } else {
                        System.out.println("Invalid score. Please enter a score between 0 and 100.");
                    }
                }
            }
        }

        System.out.println("\nClass Summary:");
        System.out.println("Students\tSub1\tSub2\tSub3\tTot\tAve\tPos");
        int[] totals = new int[numStudents];
        double[] averages = new double[numStudents];
        for (int count = 0; count < numStudents; count++) {
            int total = 0;
            for (int index = 0; index < numSubjects; index++) {
                total += scores[count][index];
            }
            totals[count] = total;
            averages[count] = (double) total / numSubjects;
            DecimalFormat df = new DecimalFormat("#.00");
            System.out.println("Student " + (count + 1) + "\t" + scores[count][0] + "\t" + scores[count][1] + "\t" + scores[count][2] + "\t" + total + "\t" + df.format(averages[count]) + "\t" + (count + 1));
        }

        System.out.println("\nSubject Summary:");
        int hardestSubject = 0;
        int easiestSubject = 0;
        int hardestSubjectFails = 0;
        int easiestSubjectPasses = 0;
        int overallHighestScore = 0;
        int overallHighestScoreStudent = 0;
        int overallHighestScoreSubject = 0;
        for (int index = 0; index < numSubjects; index++) {
            int maxScore = scores[0][index];
            int maxStudent = 0;
            int minScore = scores[0][index];
            int minStudent = 0;
            int totalScore = 0;
            int numPasses = 0;
            int numFails = 0;
            for (int count = 0; count < numStudents; count++) {
                if (scores[count][index] > maxScore) {
                    maxScore = scores[count][index];
                    maxStudent = count;
                }
                if (scores[count][index] < minScore) {
                    minScore = scores[count][index];
                    minStudent = count;
                }
                totalScore += scores[count][index];
                if (scores[count][index] >= 50) {
                    numPasses++;
                } else {
                    numFails++;
                }
            }
            double averageScore = (double) totalScore / numStudents;
            System.out.println("Subject " + (index + 1) + " highest scoring student is: Student " + (maxStudent + 1));
            System.out.println("Subject " + (index + 1) + " lowest scoring student is: Student " + (minStudent + 1));
            System.out.println("Subject " + (index + 1) + " total score is: " + totalScore);
            DecimalFormat df = new DecimalFormat("#.00");
            System.out.println("Subject " + (index + 1) + " average score is: " + df.format(averageScore));
            System.out.println("Subject " + (index + 1) + " number of passes: " + numPasses);
            System.out.println("Subject " + (index + 1) + " number of fails: " + numFails);

            if (numFails > hardestSubjectFails) {
                hardestSubject = index;
                hardestSubjectFails = numFails;
            }
            if (numPasses > easiestSubjectPasses) {
                easiestSubject = index;
                easiestSubjectPasses = numPasses;
            }
            if (maxScore > overallHighestScore) {
                overallHighestScore = maxScore;
		overallHighestScoreStudent = maxStudent;
                overallHighestScoreSubject = index;
            }
        }

        System.out.println("The hardest subject is subject " + (hardestSubject + 1) + " with " + hardestSubjectFails + " failures");
        System.out.println("The easiest subject is subject " + (easiestSubject + 1) + " with " + easiestSubjectPasses + " passes");
        System.out.println("The overall highest score is scored by student " + (overallHighestScoreStudent + 1) + " in subject " + (overallHighestScoreSubject + 1) + " scoring: " + overallHighestScore);

        System.out.println("============================================================== CLASS SUMMARY =======================================");
        int bestGraduatingStudent = 0;
        int bestGraduatingStudentScore = 0;
        int worstGraduatingStudent = 0;
        int worstGraduatingStudentScore = 0;
        int classTotalScore = 0;
        for (int count = 0; count < numStudents; count++) {
            classTotalScore += totals[count];
            if (totals[count] > bestGraduatingStudentScore) {
                bestGraduatingStudent = count;
                bestGraduatingStudentScore = totals[count];
            }
            if (totals[count] < worstGraduatingStudentScore || worstGraduatingStudentScore == 0) {
                worstGraduatingStudent = count;
                worstGraduatingStudentScore = totals[count];
            }
        }
        double classAverageScore = (double) classTotalScore / numStudents;
        DecimalFormat df = new DecimalFormat("#.00");
        System.out.println("Best graduating student is: student " + (bestGraduatingStudent + 1) + " scoring: " + bestGraduatingStudentScore);
        System.out.println("!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Worst graduating student is: student " + (worstGraduatingStudent + 1) + " scoring: " + worstGraduatingStudentScore);
        System.out.println("===============================================================");
        System.out.println("Class total score is: " + classTotalScore);
        System.out.println("Class average score is: " + df.format(classAverageScore));
        System.out.println("===============================================================");
    }
}

File: lagbajaschools.py
import decimal

def main():
    num_students = int(input("How many students do you have? "))
    num_subjects = int(input("How many subjects do they offer? "))

    scores = []
    for count in range(num_students):
        student_scores = []
        print("Entering score for student", count + 1)
        for index in range(num_subjects):
            while True:
                score = int(input("Enter score for subject " + str(index + 1) + ": "))
                if 0 <= score <= 100:
                    student_scores.append(score)
                    break
                else:
                    print("Invalid score. Please enter a score between 0 and 100.")
        scores.append(student_scores)

    print("\nClass Summary:")
    print("Students\tSub1\tSub2\tSub3\tTot\tAve\tPos")
    totals = []
    averages = []
    for count, student_scores in enumerate(scores):
        total = sum(student_scores)
        average = decimal.Decimal(total) / decimal.Decimal(num_subjects)
        totals.append(total)
        averages.append(average)
        print("Student", count + 1, "\t", student_scores[0], "\t", student_scores[1], "\t", student_scores[2], "\t", total, "\t", round(average, 2), "\t", count + 1)

    hardest_subject = 0
    easiest_subject = 0
    hardest_subject_fails = 0
    easiest_subject_passes = 0
    for index in range(num_subjects):
        num_passes = 0
        num_fails = 0
        for count in range(num_students):
            if scores[count][index] >= 50:
                num_passes += 1
            else:
                num_fails += 1
        if num_fails > hardest_subject_fails:
            hardest_subject = index
            hardest_subject_fails = num_fails
        if num_passes > easiest_subject_passes:
            easiest_subject = index
            easiest_subject_passes = num_passes

    overall_highest_score = 0
    overall_highest_score_student = 0
    overall_highest_score_subject = 0
    for count in range(num_students):
        for index in range(num_subjects):
            if scores[count][index] > overall_highest_score:
                overall_highest_score = scores[count][index]
                overall_highest_score_student = count
                overall_highest_score_subject = index

    best_graduating_student = 0
    best_graduating_student_score = 0
    worst_graduating_student = 0
    worst_graduating_student_score = 0
    for count in range(num_students):
        if totals[count] > best_graduating_student_score:
            best_graduating_student = count
            best_graduating_student_score = totals[count]
        if totals[count] < worst_graduating_student_score or worst_graduating_student_score == 0:
            worst_graduating_student = count
            worst_graduating_student_score = totals[count]

    print("\nThe hardest subject is subject", hardest_subject + 1, "with", hardest_subject_fails, "fails")
    print("The easiest subject is subject", easiest_subject + 1, "with", easiest_subject_passes, "passes")
    print("The overall highest score is scored by student", overall_highest_score_student + 1, "in subject", overall_highest_score_subject + 1, "scoring:", overall_highest_score)
    print("\nBest graduating student is: student", best_graduating_student + 1, "scoring:", best_graduating_student_score)
    print("Worst graduating student is: student", worst_graduating_student + 1, "scoring:", worst_graduating_student_score)


print("The hardest subject is subject", hardest_subject + 1, "with", hardest_subject_fails, "failures")
print("The easiest subject is subject", easiest_subject + 1, "with", easiest_subject_passes, "passes")
print("The overall highest score is scored by student", overall_highest_score_student + 1, "in subject", overall_highest_score_subject + 1, "scoring:", overall_highest_score)
print("The overall highest score is scored by student", overall_highest_score_student + 1, "in subject", overall_highest_score_subject + 1, "scoring:", overall_highest_score)

print("\n============================================================== CLASS SUMMARY =======================================")
print("Best graduating student is: student", best_graduating_student + 1, "scoring:", best_graduating_student_score)
print("=============================================== !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!")
print("Worst graduating student is: student", worst_graduating_student + 1, "scoring:", worst_graduating_student_score)
print("!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!")
print("=============================================")
class_total_score = sum(totals)
class_average_score = decimal.Decimal(class_total_score) / decimal.Decimal(num_students)
print("Class total score is:", class_total_score)
print("Class average score is:", round(class_average_score, 2))
print("=============================================")

if __name__ == "__main__":
    main()


